### ISSUE TITLE
Provide mechanism for checking if custom column encryption provider has been registered

### ISSUE THREAD

**Issue by DigiBanks99** [dotnet/SqlClient#1642]
### Is your feature request related to a problem? Please describe.

We make use of the `SqlColumnEncryptionAzureKeyVaultProvider` which we register as part of our host configuration. For integration testing altering the client, configuration causes a `System.InvalidOperationException` to be thrown if a second attempt is made to register the same provider. Currently, we are catching this specific instance and ignoring the exception.

### Describe the solution you'd like

Ideally having a mechanism to check if the provider has been registered would help us decide if the registration should be done again.

### Describe alternatives you've considered

I don't really know if there is another more appropriate alternative as we don't want to alter the current functionality.

### Additional context

[The exception is thrown here](https://github.com/dotnet/SqlClient/blob/d600f8921e193c72b37d728b8a76cee2cfece9ac/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs#L311)

**Comment by roji** [comment]
Shouldn't this kind of thing be tracked in your integration testing infrastructure, and you'd just avoid registering the same provider twice?

**Comment by DigiBanks99** [comment]
Well, since the logic to track registration lives entirely inside the `SqlConnection` class with no mechanism to tap into it, I don't see how I'm supposed to do that?

I can modify my ASP.NET app with a static field. But this seems very extreme to make a change to a running app to track something a different dependency is doing simply so test will pass, don't you think? The fact `SqlConnection` tracks this in a static means feels to me like the source of truth lies with it.

**Comment by roji** [comment]
> Well, since the logic to track registration lives entirely inside the SqlConnection class with no mechanism to tap into it, I don't see how I'm supposed to do that?

I'm not sure what the problem is - it's your responsibility to register this once before opening connections, so you'd do this in your test infrastructure in some single setup code which gets executed once, before tests run. In an ASP.NET app, simply run this in your application's Main or similar...

If for whatever reason that doesn't work for you, have some static bool which you set to know whether you've already registered. That's probably going to be less code and easier to do than to access some SqlClient API for getting all registered providers and checking if one of them is what you're looking for.

**Comment by roji** [comment]
I'm not a SqlClient maintainer, so it's not up to me - adding a Try method that returns a bool rather than throwing may be acceptable. But it's a fairly common scenario to have to run some initialization code once in the context of an application, and IMHO it doesn't seem like much to ask an application (or test) writer to handle that.

**Comment by JRahnama** [comment]
@DigiBanks99 just to clarify are you asking for a new feature to be added to our driver? If the answer is yes, that would be a low priority in our queue of to-do tasks as it could, as @roji  mentioned, be handled in the user application.

**Comment by DigiBanks99** [comment]
> @DigiBanks99 just to clarify are you asking for a new feature to be added to our driver? If the answer is yes, that would be a low priority in our queue of to-do tasks as it could, as @roji mentioned, be handled in the user application.

@JRahnama Yes, it is indeed a new feature request. I understand what you mean, so would you be open to a PR?

**Comment by JRahnama** [comment]
@DigiBanks99 as an open source driver we always appreciate contribution from users and community members. Please feel free to submit a PR that we can review that.
